x-pack/plugins/actions/server/lib/connector_metrics_collector.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AxiosError, AxiosResponse } from 'axios';
+
+interface ConnectorMetrics {
+  requestBodyBytes: number;
+}
+
+export class ConnectorMetricsCollector {
+  private metrics: ConnectorMetrics = {
+    requestBodyBytes: 0,
+  };
+
+  public addRequestBodyBytes(result?: AxiosError | AxiosResponse, body: string | object = '') {
+    const sBody = typeof body === 'string' ? body : JSON.stringify(body);
```

Comment:
I don't have a great feeling about depending on axios to send a `string | object`.  Or, I'm afarid of using `JSON.stringify()` without a try/catch - it **__can__** throw :-).  I'd be ok with using a `${body}` version of the content length in case of an error.

Since this could be expensive, I think we should move this AFTER the `Content-Length` check, since that seems highly likely to be the common case.  So just check if Content-Length is there and return that, otherwise do the other calculations (stringify, byteLength, etc).

Reply:
It's not axios returning `string | object`, it's us passing the `body` as a fallback here, and we always pass it either as string or object. And breaking JSON.stringify is really hard, it just stringify what ever you pass :)

But you are right putting it after  `Content-Length` check is better. 
Moved it and wrapped with try-catch just in case.